source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
 
-   equivalent to "file"
+================  ========
+freshstarttail:   "on|off"
+Default:          "off"
+Available since:  8.18.0
+================  ========
 
-.. index:: 
-   single: imfile; $InputFileTag
-.. function:: $InputFileTag tag:
+If set to *off*, rsyslog will read files from *last read position* as defined in State files.
+If set to *on*, rsyslog will seek the file reading position at the end/tail, processing new events written since that moment.
+It's very useful when deploying rsyslog to a large number of servers, because it allows to discard old logs.
 
-   equivalent to: "tag"
-   you would like to see the colon after the tag, you need to specify it
-   here (as shown above).
+=====================  ==============================
+persiststateinterval:  "number"
+Default:               "0 (end rsyslog execution)"
+Legacy:                $InputFilePersistStateInterval
+=====================  ==============================
 
-.. index:: 
-   single: imfile; $InputFileStateFile
-.. function:: $InputFileStateFile name-of-state-file
+Specifies how often (lines read) the state file shall be written when processing the input file. 
 
-   equivalent to: "StateFile"
+If set to 0, it means a new state file is only written when the monitored files are closed (end of rsyslogd execution).
+Any other value means that the state file is written when *number* lines have been processed.
 
-.. index:: 
-   single: imfile; $InputFileFacility
-.. function:: $InputFileFacility facility
+This setting can be used to guard against message duplication due to fatal errors (like power fail).
+Note that this setting affects imfile performance, especially when set to a low value. 
 
-   equivalent to: "Facility"
+Multiline parameters
+====================
 
-.. index:: 
-   single: imfile; $InputFileSeverity
-.. function:: $InputFileSeverity severity
+=========  =======
+readmode:  "0|1|2"
+Default:   "0"
+=========  =======
 
-   equivalent to: "Severity"
+Provides support for *simple* multi-line processing. Incompatible with *startmsg.regex*. 
 
-.. index:: 
-   single: imfile; $InputRunFileMonitor
-.. function:: $InputRunFileMonitor
+* 0 - line: each line is a new message.
+* 1 - paragraph: there is a blank line between log messages.
+* 2 - indented: new messages start at the beginning of a line. If a line starts with a space or tab ("\\t") it is part of the previous message.
 
-   This **activates** the current monitor. It has no parameters. If you
-   forget this directive, no file monitoring will take place.
+================  =================
+startmsg.regex:   "POSIX ERE regex"
+Default:          none
+Available since:  8.10.0
+================  =================
 
-.. index:: 
-   single: imfile; $InputFilePollInterval
+This option allows more complex multi-line messages, using provided regex as a start of new messages. As it is using regular expressions, it's more flexible than *readMode* but at the cost of lower performance.
```

Comment:
per Rainer, this is not a significant performance impact if the regex is simple. we should test this with the readmode-equivalent regexes and if it really is the same (or very close) change the performance comment to be that complex regexes can be expensive rather than saying that this is slower than readmode